File: VB/ASPxGridView_Menu/App_Code/Invoice.vb
Imports System
Imports System.Collections.Generic

<Serializable> _
Public Class Invoice
    Implements IEquatable(Of Invoice)


    Private id_Renamed As Integer

    Public Property Id() As Integer
        Get
            Return id_Renamed
        End Get
        Set(ByVal value As Integer)
            id_Renamed = value
        End Set
    End Property

    Private description_Renamed As String

    Public Property Description() As String
        Get
            Return description_Renamed
        End Get
        Set(ByVal value As String)
            description_Renamed = value
        End Set
    End Property

    Private price_Renamed As Decimal

    Public Property Price() As Decimal
        Get
            Return price_Renamed
        End Get
        Set(ByVal value As Decimal)
            price_Renamed = value
        End Set
    End Property

    Public Sub New(ByVal id As Integer, ByVal description As String, ByVal price As Decimal)
        Me.Id = id
        Me.Description = description
        Me.Price = price
    End Sub
    Public Sub New()

    End Sub
    Public Function Equals(ByVal other As Invoice) As Boolean Implements IEquatable(Of Invoice).Equals
        Return Me.Id.Equals(other.Id)
    End Function
End Class

File: CS/ASPxGridView_Menu/App_Code/Invoice.cs
using System;
using System.Collections.Generic;

[Serializable]
public class Invoice : IEquatable<Invoice> {
    private int id;

    public int Id {
        get { return id; }
        set { id = value; }
    }
    private string description;

    public string Description {
        get { return description; }
        set { description = value; }
    }
    private decimal price;

    public decimal Price {
        get { return price; }
        set { price = value; }
    }

    public Invoice(int id, string description, decimal price) {
        this.Id = id;
        this.Description = description;
        this.Price = price;
    }
    public Invoice() {

    }
    public bool Equals(Invoice other) {
        return this.Id.Equals(other.Id);
    }
}